stl/src/fiopen.cpp

```diff
@@ -40,6 +40,14 @@ namespace {
             ios_base::in | ios_base::out | ios_base::binary,
             ios_base::in | ios_base::out | ios_base::trunc | ios_base::binary,
             ios_base::in | ios_base::out | ios_base::app | ios_base::binary,
+#ifdef __cpp_lib_ios_noreplace
+            ios_base::out | ios_base::noreplace,
+            ios_base::out | ios_base::trunc | ios_base::noreplace,
+            ios_base::out | ios_base::in | ios_base::trunc | ios_base::noreplace,
+            ios_base::out | ios_base::binary | ios_base::noreplace,
+            ios_base::out | ios_base::binary | ios_base::trunc | ios_base::noreplace,
+            ios_base::out | ios_base::in | ios_base::trunc | ios_base::binary | ios_base::noreplace,

```

Comment:
Browsing through the UCRT source, it looks like this is a documentation issue. `_fsopen` is a wrapper around `_openfile`, which will translate `'x'` to `_O_EXCL`. This eventually calls down to `_wsopen_nolock`, which translates `_O_EXCL` to `OPEN_EXISTING` for `CreateFileW`. If this is right, I think it would be better to use that flag, not try to check for file existance ourselves, and avoid the TOCTOU issue.

Reply:
I pushed changes, should be resolved now, hopefully, if anything else is needed/discovered, please feel free to ping me on discord.